vividus-allure-adaptor/src/main/java/org/vividus/report/allure/plugin/VisualPlugin.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.report.allure.plugin;
+
+import static io.qameta.allure.entity.TestResult.comparingByTimeAsc;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.SerializationUtils;
+
+import io.qameta.allure.CommonJsonAggregator;
+import io.qameta.allure.core.LaunchResults;
+import io.qameta.allure.entity.Attachment;
+import io.qameta.allure.entity.Status;
+import io.qameta.allure.entity.Step;
+import io.qameta.allure.entity.TestResult;
+import io.qameta.allure.tree.DefaultTreeLayer;
+import io.qameta.allure.tree.TestResultGroupFactory;
+import io.qameta.allure.tree.TestResultLeafFactory;
+import io.qameta.allure.tree.TestResultTree;
+import io.qameta.allure.tree.TestResultTreeGroup;
+import io.qameta.allure.tree.TestResultTreeLeaf;
+import io.qameta.allure.tree.Tree;
+import io.qameta.allure.tree.TreeLayer;
+
+public class VisualPlugin extends CommonJsonAggregator
```

Comment:
Allure report should "know" nothing about visual tests, in other words such plugins should be moved to the corresponding plugins